server/drivers/hd44780.c

```diff
@@ -112,6 +112,164 @@ unsigned char HD44780_scankeypad(PrivateData *p);
+	{ "hd66710",      HD44780_MODEL_EXTENDED },
+
+	{ "winstar_oled", HD44780_MODEL_WINSTAR_OLED },
+	{ "weh00xxyya",   HD44780_MODEL_WINSTAR_OLED },
+
+	{ "pt6314_vfd",   HD44780_MODEL_PT6314_VFD },
+
+	{ "",             HD44780_MODEL_DEFAULT }
+};
+
+static int model_by_name( const char *name )
+{
+	int i;
+
+	for (i=0; i<sizeof(model_mapping)/sizeof(model_mapping[0]); i++) {
+		if (strcasecmp(model_mapping[i].name, name) == 0 )
+			return model_mapping[i].model;
+	}
+
+	return -1;
+}
+
+static const char *model_name( int type )
+{
+	int i;
+
+	for (i=0; i<sizeof(model_mapping)/sizeof(model_mapping[0]); i++) {
+		if (model_mapping[i].model == type)
+			return model_mapping[i].name;
+	}
+
+	return "";
+}
+
+static const struct BacklightValueMapping {
+	const char *name;
+	int value;
+} bl_value_mapping[] = {
+	/* these below can be combined with comma */
+	{"external",     BACKLIGHT_EXTERNAL_PIN },
+	{"internal",     BACKLIGHT_INTERNAL     },
+	{"internalCmds", BACKLIGHT_CONFIG_CMDS },
+
+	/* special marker used below */
+	{ "",            BACKLIGHT_NONE },
+
+	{"none",         BACKLIGHT_NONE },
+
+	/* compability stuff for old boolean option */
+	{"0",            BACKLIGHT_NONE },
+	{"n",            BACKLIGHT_NONE },
+	{"no",           BACKLIGHT_NONE },
+	{"off",          BACKLIGHT_NONE },
+	{"false",        BACKLIGHT_NONE },
+
+	{"1",            BACKLIGHT_EXTERNAL_PIN },
+	{"y",            BACKLIGHT_EXTERNAL_PIN },
+	{"yes",          BACKLIGHT_EXTERNAL_PIN },
+	{"on",           BACKLIGHT_EXTERNAL_PIN },
+	{"true",         BACKLIGHT_EXTERNAL_PIN }
+
+};
+
+/* parses null-terminated option value for backlight handling and returns bitmask of enabled values
+ * Returns -1 if value is not valid */
+static int parse_backlight_option( const char *value )
+{
+	int result = 0;
+	int i;
+	int any = 0;
+
+	/* first find just whole option name */
+	for (i=0; i<sizeof(bl_value_mapping)/sizeof(bl_value_mapping[0]); i++) {
+		if (strcasecmp(value, bl_value_mapping[i].name) == 0)
+			return bl_value_mapping[i].value;
+	}
+
+	/* now find possibly all instances */
+	while (*value != '\0') {
+		const char *eot = strpbrk(value, ", |");
+		if (!eot)
+			eot = value + strlen(value);
+
+		i = 0;
+		for (i=0; i<sizeof(bl_value_mapping)/sizeof(bl_value_mapping[0]); i++) {
+			size_t len;
+
+			/* break loop when marker found*/
+			if (bl_value_mapping[i].name[0] == '\0')
+				/* found unknown value name - so invalid format */
+				return -1;
+
+			len = strlen(bl_value_mapping[i].name);
+			if (len != eot - value)
+				continue;
+
+			if( strncasecmp(value, bl_value_mapping[i].name, len) == 0) {
+				result |= bl_value_mapping[i].value;
+				any = 1;
+				break;
+			}
+		}
+
+		value = eot;
+		while (*value == ' ' || *value == '|' || *value == ',')
+			value ++;
+
+		/* try next one if present */
+	}
+
+	if (any) {
+		return result;
+	}
+
+	return -1;
+}
+
+static const char *backlight_type_str(int backlight_type)
+{
+	static char buffer[256] = "";
```

Comment:
The lcdproc driver API disallows static (global and local) variables in drivers. Maybe it's easiest to just drop the logging code using this function. Or turn it into a debug output and just log the numeric value...

Reply:
Yes, you are right. I just change getting textual of option value to report function and use buffer on stack.